File: RSF To Map_.Net/Upgraded/modGlobals.cs
using Microsoft.VisualBasic;
using System;
using System.Text;
using UpgradeHelpers.Helpers;

namespace RsfToMap
{
	internal static class modGlobals
	{

		internal static string[] Tokenize(ref string pData)
		{
			StringBuilder CurWord = new StringBuilder(); //current token
			string[] WL = null; //array of tokens
			string ch = ""; //current character
			string SepChar = ","; //separation character //make sepchar a comma
			WL = new string[]{""};
			int Num = -1; //number of tokens
			pData = pData + SepChar; //add a comma to the end of the string, to
			// make sure the last word/token is stored
			int tempForEndVar = Strings.Len(pData);
			for (int t = 1; t <= tempForEndVar; t++)
			{
				ch = pData.Substring(t - 1, Math.Min(1, pData.Length - (t - 1)));
				if (ch == SepChar)
				{
					if (CurWord.ToString() != "")
					{ //save word, start a new one
						Num++;
						WL = ArraysHelper.RedimPreserve(WL, new int[]{Num + 1});
						WL[Num] = CurWord.ToString(); //save word to list
						CurWord = new StringBuilder(""); //reset CurWord to empty string
					}
				}
				else
				{
					CurWord.Append(ch); //CH is not a comma, so add it to the
					//current word...
				}
			}
			return WL; //returns an array of strings
		}

		internal static bool FileExists(string filename)
		{
			return Strings.Len(FileSystem.Dir(filename, FileAttribute.Normal)) > 0;
		}
	}
}

File: RSF To Map/VBto VBNET/modGlobals.vb
'Option Strict Off
'Option Explicit On

Imports Microsoft.VisualBasic

Public Module modGlobals


	'=========================================================
#Const IndexCurFile = 3
#If CompileAll Or CompileAllBAS Or ((IndexCurFile >= CompileFilesIndexMin) And (IndexCurFile <= CompileFilesIndexMax)) Then
#Const CompileAll_modGlobals = True
#Else
'#Const CompileAll_modGlobals = True
#End If
    ' VBto upgrade warning: 'Return' As Variant --> As String
    Public Function Tokenize(ByRef pData As String) As String()
        Tokenize = Nothing
'#Const Compile_Tokenize = True
#If Compile_Tokenize Or CompileAll_modGlobals Then
        Dim Num As Short 'number of tokens
        Dim CurWord As String = "" 'current token
        Dim WL() As String 'array of tokens
        Dim ch As String 'current character
        Dim SepChar As String 'separation character
        Dim t As Integer
        SepChar = "," 'make sepchar a comma
        ReDim WL(0)
        Num = -1
        pData += SepChar 'add a comma to the end of the string, to
        ' make sure the last word/token is stored
        For t = 1 To Len(pData)
            ch = pData(t - 1).ToString()
            If ch=SepChar Then
                If CurWord<>"" Then 'save word, start a new one
                    Num += 1
                    ReDim Preserve WL(Num)
                    WL(Num) = CurWord 'save word to list
                    CurWord = "" 'reset CurWord to empty string
                End If
            Else
                CurWord += ch 'CH is not a comma, so add it to the
                'current word...
            End If
        Next t
        Tokenize = WL 'returns an array of strings
#End If	' Compile_Tokenize
    End Function

    Public Function FileExists(ByVal filename As String) As Boolean
        FileExists = False
'#Const Compile_FileExists = True
#If Compile_FileExists Or CompileAll_modGlobals Then
        FileExists = Len(Dir(filename, FileAttribute.Normal))>0
#End If	' Compile_FileExists
    End Function


End Module